### PR TITLE
Enable testing on Travis with basic pep8 check on checkin2015 dir.

### PR DESCRIPTION
**Pull request by benpatterson**
Hi - I saw this project on [volunteermatch](https://www.volunteermatch.org/search/opp1689203.jsp)

Hi!

This is a first cut at addressing #13, although it's not on Jenkins (which would need to be hosted, etc). Instead this uses Travis for free as an open source project.

Rather than solving all pep8 issues in this PR, it is addressing one directory to see if this is something you'd be interested in. I am guessing deployment/etc can be sorted out with some Travis options (there are deployment/post-build options), or as a separate thing (e.g., using heroku, etc...really depends on how you'd want to do it.) But because there are myriad options I did not take the time to address all pep8 issues or write tests, and I did not look at deployment; instead this is just a first cut to see if it's something that would work for this project.

### CODE REVIEW

**Comment by benpatterson** [comment]
BTW only an admin on this repo can turn Travis on. I have enabled it on my fork. You can see a PR against my fork's master here: https://github.com/benpatterson/checkin2015/pull/1

File: leaderboard/views.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 def calculate_rankings(company_dict):
     ranks = {}
-    ranks['percent_green_commuters'], ranks['percent_participation'], ranks['percent_green_switches'], ranks['percent_healthy_switches'], ranks['percent_avg_participation'] = [],[],[],[],[]
+    ranks['percent_green_commuters'], ranks['percent_participation'], ranks['percent_green_switches'], ranks['percent_healthy_switches'], ranks['percent_avg_participation'] = [], [], [], [], []
```
Comment (benpatterson) [inline leaderboard/views.py:17]: author's note: I know this wouldn't have been tested; I was attempting to kick a build. Then realized I needed to do it on my fork.

**Comment by benpatterson** [comment]
Oh obviously you're using openshift. Here's the [Travis support](https://docs.travis-ci.com/user/deployment/openshift/) for that.

**Comment by pwnage101** [comment]
Thanks for your work on this!  I'm not extremely familiar with github/travis-ci so I will be catching up so I can verify these changes.

**Comment by benpatterson** [comment]
Yeah basically this pull request is saying "how about Travis?" Pep8 is really being used as a starter example.

I'm happy to help build out some test infrastructure here for starters (i.e. start building out tests and ensuring those run on pull requests)...I was just looking for a CI to do that against. I think Travis could do that pretty well but there are other tools like circle-ci or codeship that I'm less familiar with personally that could be viable alternatives.

But anyway thanks for the look.

**Comment by pwnage101** [comment]
By the looks of it, travis is a 3rd party service which:
- pulls and runs tests against our code (e.g. pep8)
- can optionally deploy to openshift (e.g. deploy to a development "openshift app" for manual or automatic regression tests)

I'm totally open to using it for the former.  Regarding the latter, It's not clear to me how to make travis deploy using non-personal openshift credentials (which is not just important for security, but important for being a reliable solution).

In any case, I think we should incorporate it.  I guess it's free for us because the project is open source?

**Comment by benpatterson** [comment]
Yes...free for open source. Hurrah! (The limit is 5 concurrent builds. If that limit is encountered, then newer builds wait for a node to finish.)

It's possible to encrypt sensitive data in the yml or store it with environment variables you can set in the Travis admin panel.

But...yeah, using Travis for tests-only is totally fine and how I've generally used it, personally.

**Comment by benpatterson** [comment]
If you go this route you'd need to enable it via Travis config. You could attempt that first to see what kind of github permissions travis needs.

**Pull request merged**